addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -370,6 +390,15 @@ private void disconnect() {
         logger.trace("Squeeze Server connection stopped.");
     }
 
+    private void updateChannel(String channelID, State state) {
+        logger.trace("Updating channel {} for {} to state {}", channelID, getThing().getUID(), state);
+        try {
+            updateState(channelID, state);
+        } catch (IllegalStateException e) {
```

Comment:
In what case will this happen of did I already ask that?

Reply:
The API spec for `BaseThingHandler` says `if handler is not initialized correctly, because no callback is present`

And, I chose to not do this. ;-)
https://github.com/openhab/openhab2-addons/blob/master/addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxPlayerHandler.java#L471

Reply:
Hmm, I'm not happy with this, @sjka  @triller-telekom do you have any useful suggestions here.

Reply:
These [lines](https://github.com/eclipse/smarthome/blob/7717a6de4b4bd68eb0b6d02f7312ed2eb6ee57f0/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseThingHandler.java#L343-L347) are in the `BaseThingHandler` since 17.03.2015. So this Exception was thrown ever since then. Since it is a `RuntimeException` it does not have to be caught at compile time. And since all the other bindings do not catch it at compile time I think it's alright to let it pass through to the code above.

Reply:
Ok. I'll remove it from here and `SqueezeBoxPlayerHandler`.

Reply:
Did you do that already?

Reply:
Duh. It would help if I actually pushed the commit.